README.md

```diff
@@ -1,77 +1,64 @@
+|                                                  |
+├──────────────────────────────────────────────────┤
+|                                                  |
+|                   Block Layer                    |
+|               (cid, data, codec)                 |           
+|                                                  |
+└──────────────────────────────────────────────────┘
 ```
 
-The goal of this stack is to enable decentralized data-structures
-which in turn will enable more decentralized applications.
+## Block Layer (Layer 0)
 
-Many of the specifications in this stack are inter-dependent.
+The block layer encompasses all content addressed block formats and specifies how blocks
+are addressed,how they self-describe their codec for encoding/decoding, and how block link
+between each other.
 
-```
-                          IPLD Dependency Graph
-
-┌─────┐                           ┌────────────┐
-│ CID ├───────────┬───────────────> Raw Blocks │
-└─────┘           │               └─┬──────────┘
-          ┌───────v──────────────┐  │
-┌──────┐  │ Links (Conceptually) │  │
-│ Path │  └───────┬──────────────┘  │           ┌─────────────┐
-└──┬───┘          │                 ├───────────> Replication │
-   │              │                 │           └─────────────┘
-┌──v──────────────v─────────────────┴──────┐
-│                  Codecs                  │
-│  ┌─────┐   ┌───────────────────────────┐ │ ┌────────────────────────────┐
-│  │ Git │   │       Data Model v1       │ │ │  Multi-block Collections   │
-│  └─────┘   │ ┌──────────┐ ┌──────────┐ │ │ │ ┌────────────────────────┐ │
-│ ┌────────┐ │ │ dag-json │ │ dag-cbor │ ├─┼─> │    Specialized Data    │ │
-│ │  BTC   │ │ └──────────┘ └──────────┘ │ │ │ │ Structures & Utilities │ │
-│ └────────┘ └───────────────────────────┘ │ │ └────────────────────────┘ │
-│ ┌────────┐ ┌────────┐                    │ └───┬─────────┬──────────────┘
-│ │ Zcash  │ │ dag-pb │                    │     │         │
-│ └────────┘ └────────┘                    │     │         │
-└──────────────────────────────────────────┘     │         │
-                                                 │         │
-               ┌──────────────────────┐          │         │
-               │ File System (UnixFS) <──────────┘         │
-               └──────────────────────┘                    │
-               ┌───────────────────────┐                   │
-               │ End-User Applications <───────────────────┘
-               └───────────────────────┘
+This layer alone is enough to accomplish basic graph replication across a variety of formats (eth, bitcoin, git, etc).
 
-```
+This layer does not define data structures or types, although many codecs may convert these formats into native types,
+there are no type requirements or assurances about types at the block layer.
+
+## Data Model Layer (Layer 1)
+
+The Data Model Layer describes a set of base required types to be implemented by a subset of IPLD codecs.
```

Comment:
I don't understand the "by a subset of IPLD codecs" part.

Reply:
What I mean is, “not all IPLD codecs support the full data model.”

Reply:
I've been flirting with the idea of defining it as:

- "IPLD native formats" can represent any tree of data in the IPLD Data Model.  This basically describes dag-cbor.
- "IPLD adjunct formats" can be mapped onto the IPLD Data Model, and we can do some operations on them through that lens, but they *can't* support *any arbitrary* data from the IPLD Data Model being shoveled into them.  And this describes git, eth, etc.

Reply:
I like “native” but I’m not a huge fan of “adjunct.”

But I’m wondering if we can get rid of this entire distinction. It’s the role of a codec to **present its internal format in IPLD Data Model kinds**. Even if it doesn’t support the entire Data Model the codec needs to conform to it.

Then, the distinction for a **native** codec is one that “can encode/decode data in all IPLD Data Model kinds.”

Reply:
Intensely thumbs up to that text in bold.

(And I'm not particularly fond of 'adjunct' as a term either.)